src/Functions/FunctionsExternalDictionaries.h

```diff
@@ -464,6 +484,19 @@ class FunctionDictGetNoType final : public IFunction
         }
         else
         {
+            if (dictionary_get_function_type == DictionaryGetFunctionType::getAll && current_arguments_index < arguments.size())
+            {
+                auto limit_col = arguments[current_arguments_index].column;
+                if (!limit_col || !isColumnConst(*limit_col))
```

Comment:
Do we need to check if it is uint as well?

Reply:
I don't think so since the call to `getUInt` right after this already throws an exception if we can't convert to a uint. It's also nice because `getUInt` will cast other arithmetic types to uint, so it still works if I pass other integer types besides uint64.

I added a comment; let me know if that's insufficient.